apps/portal/components/schedule/customer-form/index.js

```
@@ -76,26 +76,21 @@ const validateCouponForm = async (props, userInputs) => {
 const validateCreditCardForm = async props => {
   if (!props.lead && !props.lead.id) return { isValid: false }
   let newState = {}
-  const stripeData = await createStripeToken({})
-  if (!stripeData.error) {
-    try {
-      await gql.request(
-        `
-          mutation {
-            createClientFromLead(
-              leadId: "${props.lead.id}"
-              stripeSource: "${stripeData.token.id}"
-            ) {
-              name
-            }
+  const { token, error } = await createStripeToken({})
+  if (error) throw error
+  await gql.request(
+    `
+        mutation {
+          createClientFromLead(
+            leadId: "${props.lead.id}"
+            stripeSource: "${token.id}"
+          ) {
+            name
           }
-        `
-      )
-      newState.isValidCard = true
-    } catch (err) {
-      throw new Error(err)
```

Try to avoid `catching` where you don't need to. This line, for instance, is doing the exact same thing as if you didn't wrap it in a `try/catch` in the first place. I noticed in the lead-form you're [catching the error and returning it instead](https://github.com/findOctave/cortex/blob/master/apps/portal/components/schedule/lead-form/index.js#L52) too. This ends up swallowing the error and causes hard to trace bugs (e.g. if someone enters an invalid phone number it just silently fails).